### PR TITLE
LUD-133 enable PCI passthrough on VM

### PR DESCRIPTION
**Pull request by danielseankim**
For ScaleIO cluster, each host (with scale io enabled) will need
a SVM that has PCI passthrough for the HBA device. This commit,
along with LUD-37, enable that requirement. This commit focuses
primarily on VM configuration. This is also a pre-req for LUD-134.

### CODE REVIEW

**Comment by danielseankim** [comment]
@sushilrai can you review / merge this please?

File: lib/puppet/provider/vc_vm/default.rb [hunk 1]
```diff
@@ -754,6 +762,51 @@ def vm_config_spec(path="")
...
+    vendor_id = pci_device.vendorId
+    host_uuid = find_vm_host.esxcli.system.uuid.get
+
+    Puppet.debug("Configuring PCI passthrough on VM with device, #{pci_device}: id=#{pci_id}, device_id=#{pci_device_id}, vendor_id=#{vendor_id}, host_uuid=#{host_uuid}")
+    backing = RbVmomi::VIM.VirtualPCIPassthroughDeviceBackingInfo(
+        :id => pci_id,
+        :deviceId => pci_device_id,
+        :vendorId => vendor_id,
+        :systemId => host_uuid,
+        :deviceName => ""
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm/default.rb]: Do we need to send blank deviceName to `VirtualPCIPassthroughDeviceBackingInfo` ?

File: lib/puppet/provider/vc_vm/default.rb [hunk 2]
```diff
@@ -754,6 +762,51 @@ def vm_config_spec(path="")
...
+
+    pciDevice = RbVmomi::VIM.send(:VirtualPCIPassthrough, :backing => backing, :key => 0)
+
+    RbVmomi::VIM.VirtualDeviceConfigSpec(
+        :device => pciDevice,
+        :operation => RbVmomi::VIM.VirtualDeviceConfigSpecOperation('add')
+    )
+  end
+
+
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm/default.rb]: Remove extra white spaces

**Comment by sushilrai** [comment]
Changes looks good. Just one question

File: lib/puppet/provider/vc_vm/default.rb (hunk 1 above)
Comment (danielseankim) [reply lib/puppet/provider/vc_vm/default.rb]: @sushilrai this is just due to the vmware api. I created a VM with PCI passthrough for the RAID controller, and it had the same setting. I believe the API call causes error for missing required field if you don't supply deviceName.

**Pull request merged**